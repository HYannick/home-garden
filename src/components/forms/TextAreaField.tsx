/** @jsx jsx */
import React, { Fragment } from 'react';
import styled from '@emotion/styled';
import { jsx } from '@emotion/core';
import { Label } from './Label';

interface FieldProps {
  field?: any,
  form?: any,
  placeholder?: string,
  label?: string
}

const TextArea = styled('textarea')`
  border: 0.3rem solid ${({ theme }) => theme.palette.grey.light};
  outline: ${({ theme }) => theme.palette.grey.light};
  padding: 1rem 2rem;
  border-radius: 1.5rem;
  width: 100%;
  font-size: 1.6rem;
  min-height: 20rem;
`;


const TextAreaField: React.FC<FieldProps> = ({ field, label }) => (
  <Fragment>
    <Label htmlFor={field.name}>{label}</Label>
    <TextArea id={field.name} name={field.name} onChange={field.onChange} />
  </Fragment>
);

export default TextAreaField;
