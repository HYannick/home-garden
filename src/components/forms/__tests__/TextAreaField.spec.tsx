import React from 'react';
import 'jest-dom/extend-expect';
import { cleanup, fireEvent } from '@testing-library/react';
import { renderWithTheme } from '../../../theme-wrapper';
import TextAreaField from '../TextAreaField';

afterEach(cleanup);
describe('TextAreaField', () => {
  const props = {
    field: {
      name: 'puke',
      onChange: jest.fn(),
    },
    type: 'text',
    label: 'Should I puke?',
  };
  it('should render properly', () => {
    const { container } = renderWithTheme(
      <TextAreaField {...props}/>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should render the right label', () => {
    const { getByText } = renderWithTheme(
      <TextAreaField {...props}/>,
    );
    expect(getByText(/^Should I puke/)).toHaveTextContent('Should I puke');
  });
  it('should trigger onChange', () => {
    const { getByLabelText } = renderWithTheme(
      <TextAreaField {...props}/>,
    );
    const input = getByLabelText('Should I puke?');
    fireEvent.change(input, { target: { value: 'm' } });
    expect(props.field.onChange).toHaveBeenCalledTimes(1);
  });
});
