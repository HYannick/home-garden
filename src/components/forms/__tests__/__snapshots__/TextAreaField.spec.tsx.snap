// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaField should render properly 1`] = `
<div>
  <label
    class="css-npa6bc"
    for="puke"
  >
    Should I puke?
  </label>
  <textarea
    class="css-vygrie"
    id="puke"
    name="puke"
  />
</div>
`;
